lib/eslint.js

```diff
@@ -940,28 +890,33 @@ module.exports = (function() {
         return sourceCode;
     };
 
-    /**
-     * Retrieves an array containing all comments in the source code.
-     * @returns {ASTNode[]} An array of comment nodes.
-     */
-    api.getAllComments = function() {
-        return currentAST.comments;
+    // methods that exist on SourceCode object
+    var externalMethods = {
+        getSource: "getText",
+        getSourceLines: "getLines",
+        getAllComments: "getAllComments",
+        getNodeByRangeIndex: "getNodeByRangeIndex",
+        getComments: "getComments",
+        getJSDocComment: "getJSDocComment",
+        getFirstToken: "getFirstToken",
+        getFirstTokens: "getFirstTokens",
+        getLastToken: "getLastToken",
+        getLastTokens: "getLastTokens",
+        getTokenAfter: "getTokenAfter",
+        getTokenBefore: "getTokenBefore",
+        getTokenByRangeStart: "getTokenByRangeStart",
+        getTokens: "getTokens",
+        getTokensAfter: "getTokensAfter",
+        getTokensBefore: "getTokensBefore",
+        getTokensBetween: "getTokensBetween"
     };
 
-    /**
-     * Gets all comments for the given node.
-     * @param {ASTNode} node The AST node to get the comments for.
-     * @returns {Object} The list of comments indexed by their position.
-     */
-    api.getComments = astUtils.getComments;
-
-    /**
-     * Retrieves the JSDoc comment for a given node.
-     * @param {ASTNode} node The AST node to get the comment for.
-     * @returns {ASTNode} The BlockComment node containing the JSDoc for the
-     *      given node or null if not found.
-     */
-    api.getJSDocComment = astUtils.getJSDocComment;
+    // copy over methods
+    Object.keys(externalMethods).forEach(function(methodName) {
+        api[methodName] = function() {
+            return sourceCode ? sourceCode[externalMethods[methodName]].apply(sourceCode, arguments) : null;
```

Comment:
There's an odd case where `sourceCode` can be null, and in that case, we cannot throw an error. So `bind()` wouldn't work for all cases here.